### PR TITLE
vscode-extensions.ms-dotnettools.csharp: 1.25.4 -> 2.22.3

### PR DESCRIPTION
**Pull request by GGG-KILLER**
## Description of changes

Update the C# extension from v1.25.4 to v2.22.2.

A major breaking change of this is that the OmniSharp language server is no longer used but instead Roslyn's with a proprietary middleware.

Possible interested area: @NixOS/dotnet
Previous maintainer: @jraygauthier 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/editors/vscode/extensions/ms-dotnettools.csharp/default.nix [hunk 1]
```diff
@@ -1,136 +1,115 @@
...
-    sed -i \
-      -E -e 's/(this\._pipePath=[a-zA-Z0-9_]+\.join\()([a-zA-Z0-9_]+\.getExtensionPath\(\)[^,]*,)/\1require("os").tmpdir(), "'"$ext_unique_id"'"\+/g' \
-      "$PWD/dist/extension.js"
-
-    # Fix reference to uname
-    sed -i \
-      -E -e 's_uname -m_${coreutils}/bin/uname -m_g' \
-      "$PWD/dist/extension.js"
+  postPatch =
+    ''
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/vscode/extensions/ms-dotnettools.csharp/default.nix:71]:
```suggestion
  postPatch = ''
```

File: pkgs/applications/editors/vscode/extensions/ms-dotnettools.csharp/default.nix [hunk 2]
```diff
@@ -1,136 +1,115 @@
...
+          binaries =
+            darwinBins
+            ++ [
+              ".debugger/arm64/vsdbg-ui"
+              ".debugger/arm64/vsdbg"
+            ];
         };
-      }.${system} or (throw "Unsupported system: ${system}");
+      }.${system}
+        or (throw "Unsupported system: ${system}");
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/vscode/extensions/ms-dotnettools.csharp/default.nix:56]:
```suggestion
      }.${system} or (throw "Unsupported system: ${system}");
```

File: pkgs/applications/editors/vscode/extensions/ms-dotnettools.csharp/default.nix [hunk 3]
```diff
@@ -1,136 +1,115 @@
...
-          sha256 = "08406xz2raal8f10bmnkz1mwdfprsbkjxzc01v0i4sax1hr2a2yl";
-          binaries = darwinAarch64DebuggerBins ++ darwinX86DebuggerBins ++ omniSharpBins ++ razorBins;
+          arch = "darwin-arm64";
+          sha256 = "sha256-sMHJk39wOoTBcLvb4NFLNv56VzOeAaLjV9vG4qUJSR0=";
+          binaries =
+            darwinBins
+            ++ [
+              ".debugger/arm64/vsdbg-ui"
+              ".debugger/arm64/vsdbg"
+            ];
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/vscode/extensions/ms-dotnettools.csharp/default.nix:53]:
```suggestion
          binaries = darwinBins ++ [
            ".debugger/arm64/vsdbg-ui"
            ".debugger/arm64/vsdbg"
          ];
```

File: pkgs/applications/editors/vscode/extensions/ms-dotnettools.csharp/default.nix [hunk 4]
```diff
@@ -1,136 +1,115 @@
...
-      chmod +x "${bin}"
-    '')
-    vsixInfo.binaries))
-  + lib.optionalString stdenv.isLinux (lib.concatStringsSep "\n" (map
-    (bin: ''
-      patchelf_common "${bin}"
-    '')
-    vsixInfo.binaries));
+    ''
+    + (lib.concatStringsSep "\n" (map
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/vscode/extensions/ms-dotnettools.csharp/default.nix:97]:
```suggestion
        --replace 'uname -m' '${lib.getExe' coreutils "uname"} -m'
    '' + (lib.concatStringsSep "\n" (map
```

File: pkgs/applications/editors/vscode/extensions/ms-dotnettools.csharp/default.nix (hunk 2 above)
Comment (GGG-KILLER) [reply pkgs/applications/editors/vscode/extensions/ms-dotnettools.csharp/default.nix]: Fixed!

**Pull request merged**